### PR TITLE
feat(admin): cross-link request log and error dump entries

### PR DESCRIPTION
**Pull request by Oaklight**
## Summary

- **Request log ↔ Error dump cross-linking**: 4xx/5xx request log entries show a jump-to-dump button; error dump rows with `request_log_id` show a jump-to-log button. Jumps clear filters, calculate correct page offset, and highlight the target row for 10 seconds.
- **Backfill matching**: "Match Logs" button and auto-run on startup link unlinked error dumps to request log entries via timestamp (±0.1s), source/target provider, status code, and model alias resolution from gateway config.
- **API key last-used tracking**: New `last_used` column auto-migrated on startup, updated in auth hook with 5-min throttle. "Refresh Last Used" icon button backfills from request log history.
- **UI improvements**: capability badge SVG icons using `cap-badge` class, model table column width tuning (Model 19%, Type 15% center, Capabilities 23%), `data-i18n-title` support, "Server Time" label, GMT±N timezone format throughout.

## Test plan

- [ ] Deploy dev build, verify cross-link buttons appear on both request log and error dump entries
- [ ] Click jump buttons in both directions, verify tab switch + row highlight
- [ ] Click "Match Logs" on error dumps, verify unlinked entries get matched
- [ ] Verify API keys Last Used column populates after making requests
- [ ] Click refresh icon on API keys, verify backfill from request log
- [ ] Verify capability badge icons render correctly on models tab
- [ ] Verify GMT±N timezone in server clock and hover timestamps
- [ ] `make test` passes

### CODE REVIEW

File: src/llm_rosetta/observability/persistence.py [hunk 1]
```diff
@@ -343,7 +343,80 @@ def get_log_entry(self, entry_id: str) -> dict[str, Any] | None:
...
+                    "SELECT id FROM request_log "
+                    "WHERE source_provider = ? AND target_provider = ? "
+                    "AND status_code = ? "
+                    "AND abs(julianday(timestamp) - julianday(?)) * 86400 < ? "
+                    "ORDER BY abs(julianday(timestamp) - julianday(?)) "
+                    "LIMIT 1",
+                    (source, target, status, ts, window_seconds, ts),
+                ).fetchone()
+            if row:
+                self._conn.execute(
```
[the remark this answers is not in the document]
Comment (Oaklight) [reply src/llm_rosetta/observability/persistence.py:420]: Added a comment documenting the risk. The fallback is intentional for cases where model aliases diverge (e.g. `gemini-3.1-pro-preview` vs `gemini-3.1-pro`), and the 0.1s window makes false matches very unlikely in practice.

File: src/llm_rosetta/gateway/keystore.py [hunk 2]
```diff
@@ -83,10 +94,52 @@ def _refresh_cache(self) -> None:
...
+        """Validate a raw API key and return ``(key_id, context)``, or None."""
+        return self._cache.get(_hash_key(raw_key))
+
+    def touch(self, key_id: str, interval: float = 300.0) -> None:
+        """Record a last-used timestamp, throttled to one write per *interval* seconds."""
+        now = time.monotonic()
+        if now - self._last_touch.get(key_id, 0) < interval:
+            return
+        self._last_touch[key_id] = now
+        from datetime import datetime, timezone
```
[the remark this answers is not in the document]
Comment (Oaklight) [reply src/llm_rosetta/gateway/keystore.py]: Moved to module-level `from datetime import datetime, timezone` in 9b7e5aa.

File: src/llm_rosetta/gateway/admin/js/logs.js [hunk 3]
```diff
@@ -66,6 +69,18 @@ function renderLogs(entries, total) {
     }).join('');
   }
 
+  // Auto-highlight and scroll
+  if (S._highlightLogId) {
+    const hlRow = document.querySelector('#logTable tr[data-log-id="' + S._highlightLogId + '"]');
+    if (hlRow) hlRow.scrollIntoView({behavior:'smooth', block:'center'});
+    if (!S._highlightLogIdTimer) {
```
[the remark this answers is not in the document]
Comment (Oaklight) [reply src/llm_rosetta/gateway/admin/js/logs.js]: Fixed — changed all three sites to `clearTimeout(S._highlightLogIdTimer)` before setting the new timer. Covers both `renderLogs`, `jumpToRequestLog`, and `jumpToErrorDump`.

File: src/llm_rosetta/gateway/admin/js/logs.js [hunk 4]
```diff
@@ -134,8 +149,44 @@ function updateKeyFilterOptions() {
...
+          }, 300);
+        }, 200);
+      } else if (attempts < 5) {
+        setTimeout(function() { tryFind(attempts + 1); }, 500);
+      } else {
+        const section = document.getElementById('errorDumpSection');
+        if (section) section.scrollIntoView({behavior:'smooth', block:'start'});
+      }
+    }
+    setTimeout(function() { tryFind(0); }, 600);
```
[the remark this answers is not in the document]
Comment (Oaklight) [reply src/llm_rosetta/gateway/admin/js/logs.js:182]: Fair point. For now the retry pattern is acceptable since dump data is loaded eagerly (10k limit) and renders quickly. If this becomes an issue we can switch to a MutationObserver, but that adds complexity for a rare edge case.

**Comment by Oaklight** [comment]
@clementine-oaklight All items addressed in 9b7e5aa:

1. **validate() tests** — Updated `test_keystore.py` to destructure `(key_id, ctx)`. All 58 tests pass.
2. **Resource leak** — Wrapped `backfill_last_used()` in try/finally.
3. **COUNT(*) in get_log_entry** — Added docstring explaining it's only called for cross-link lookups (not hot path). Acceptable for single-entry use.
4. **Lazy datetime import** — Moved to module-level.
5. **N+1 queries** — Added comment noting intentional non-batching for small expected row counts.

**Comment by Oaklight** [comment]
@milo-oaklight All items addressed in 9b7e5aa:

1. **_capIcons per-row rebuild** — Hoisted to module-level constant.
2. **Hardcoded pageSize=30** — Now uses `LOG_LIMIT` from state.js.
3. **Highlight timer doesn't reset** — Changed to `clearTimeout` before setting new timer (all three sites).
4. **validate() breaking change** — Tests updated; noted this is internal API.
5. **import_from_config** — Not part of this PR (exists on another branch that was merged separately).
